src/com/android/calendar/settings/GeneralPreferences.kt

```diff
@@ -292,8 +292,10 @@ class GeneralPreferences : PreferenceFragmentCompat(),
                 a.recreate()
             }
             KEY_PURE_BLACK_NIGHT_MODE -> {
-                Utils.sendUpdateWidgetIntent(a)
-                a.recreate()
+                if (themePref.value == "system" && DynamicTheme.isSystemInDarkTheme(a)) {
```

Comment:
`&&` sounds strange, if it worked without my suggestion before, why would it work now?
Also, my suggestion was based on how we use pure black at other places.